### PR TITLE
Add a http handler for landing diff

### PR DESCRIPTION
**Pull request by zhenghuiwang**
plus a check to read only thrift files for parsing thrift services.

r: @uber/zanzibar-team

### CODE REVIEW

File: backend/repository/handler.go [hunk 1]
```diff
@@ -347,6 +348,45 @@ func (h *Handler) CreateDiff(w http.ResponseWriter, r *http.Request, ps httprout
 	h.WriteJSON(w, http.StatusOK, &createDiffResponse{DiffURI: resp})
 }
 
+// LandDiff lands a diff.
+func (h *Handler) LandDiff(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	type LandDiffRequest struct {
```
Comment (ChuntaoLu) [inline backend/repository/handler.go]: Shouldn't this be defined outside the function?

File: backend/repository/handler.go [hunk 2]
```diff
@@ -347,6 +348,45 @@ func (h *Handler) CreateDiff(w http.ResponseWriter, r *http.Request, ps httprout
...
+	if err != nil {
+		h.WriteErrorResponse(w, http.StatusNotFound, err)
+		return
+	}
+	repo, err := h.Manager.NewRuntimeRepository(gatewayConfig.ID)
+	if err != nil {
+		h.WriteErrorResponse(w, http.StatusInternalServerError, errors.Wrap(err, "failed to create temp runtime dirs"))
+		return
+	}
+	h.logger.Info("Landing diff...",
```
Comment (ChuntaoLu) [inline backend/repository/handler.go]: seems like only happy path is logged? do we not need log the errors?

File: backend/repository/handler.go (hunk 1 above)
Comment (zhenghuiwang) [inline backend/repository/handler.go]: I prefer defining type inside a func if the struct type is only to be used within the func, similar to a func scope variable. Parsing the request is a one-time thing. If we have multiple handlers like this, defining a type inside a func seems to be more readable.

File: backend/repository/handler.go (hunk 2 above)
Comment (zhenghuiwang) [inline backend/repository/handler.go]: Errors are logged inside the `h.WriteErrorResponse` func

Comment (ChuntaoLu) [inline backend/repository/handler.go]: 👍

File: backend/repository/handler.go (hunk 1 above)
Comment (ChuntaoLu) [inline backend/repository/handler.go]: parsing is one-time, but per request. This particular handler is probably of low volume. I wonder if this practice cause overhead for high volume requests, in which case the same type declaration repeats more often.

Regarding readability, I think it makes sense for small structs, and gets awkward for non trivial ones.

**Pull request merged**